File: swarmmanagercontroller-new/src/main/java/com/swarmbit/docker/api/client/jaxrs/filter/JsonClientFilter.java
package com.swarmbit.docker.api.client.jaxrs.filter;

import javax.ws.rs.client.ClientRequestContext;
import javax.ws.rs.client.ClientResponseContext;
import javax.ws.rs.client.ClientResponseFilter;
import javax.ws.rs.core.MediaType;
import java.io.IOException;

public class JsonClientFilter implements ClientResponseFilter {

    @Override
    public void filter(ClientRequestContext requestContext, ClientResponseContext responseContext) throws IOException {
        if (responseContext.getMediaType() != null
                && responseContext.getMediaType().isCompatible(MediaType.TEXT_PLAIN_TYPE)) {
            String newContentType = "application/json" + responseContext.getMediaType().toString().substring(10);
            responseContext.getHeaders().putSingle("Content-Type", newContentType);
        }
    }
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/common/client/jaxrs/filter/JsonClientFilter.kt
package com.swarmbit.docker.api.common.client.jaxrs.filter

import java.io.IOException
import javax.ws.rs.client.ClientRequestContext
import javax.ws.rs.client.ClientResponseContext
import javax.ws.rs.client.ClientResponseFilter
import javax.ws.rs.core.MediaType

class JsonClientFilter : ClientResponseFilter {

    @Throws(IOException::class)
    override fun filter(requestContext: ClientRequestContext, responseContext: ClientResponseContext) {
        if (responseContext.mediaType != null && responseContext.mediaType.isCompatible(MediaType.TEXT_PLAIN_TYPE)) {
            val newContentType = "application/json" + responseContext.mediaType.toString().substring(10)
            responseContext.headers.putSingle("Content-Type", newContentType)
        }
    }
}
